File: app/backend/src/main/java/eaproject/output/GetAllGradesOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.sql.Timestamp;
import java.util.ArrayList;

public class GetAllGradesOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private ArrayList<GradeProperties> gradeList;

    public ArrayList<GradeProperties> getGradeList() {
        return gradeList;
    }

    public void setGradeList(ArrayList<GradeProperties> gradeList) {
        this.gradeList = gradeList;
    }

    public static class GradeProperties implements Serializable {

        @JsonProperty
        private int id;

        @JsonProperty
        private String name;

        @JsonProperty
        private int minAge;

        @JsonProperty
        private Integer maxAge;

        @JsonProperty
        private String gender;

        @JsonProperty
        private Timestamp creationDate;

        public int getId() {
            return id;
        }

        public void setId(int id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public int getMinAge() {
            return minAge;
        }

        public void setMinAge(int minAge) {
            this.minAge = minAge;
        }

        public Integer getMaxAge() {
            return maxAge;
        }

        public void setMaxAge(Integer maxAge) {
            this.maxAge = maxAge;
        }

        public String getGender() {
            return gender;
        }

        public void setGender(String gender) {
            this.gender = gender;
        }

        public Timestamp getCreationDate() {
            return creationDate;
        }

        public void setCreationDate(Timestamp creationDate) {
            this.creationDate = creationDate;
        }
    }
}

File: app/frontend/ea-project/src/models/output/GetAllGradesOutput.js
import BaseOutput from "./BaseOutput";

export default class GetAllGradesOutput extends BaseOutput {
  constructor(gradeList = [], feedbackMessages = []) {
    super(feedbackMessages);
    this.gradeList = Array.isArray(gradeList) ? gradeList.map(item => new GetAllGradesOutput.GradeProperties(item)) : [];
  }

  static GradeProperties = class {
    constructor({id = 0, name = '', minAge = 0, maxAge = 0, gender = '', creationDate = new Date()}) {
      this.id = id;
      this.name = name;
      this.minAge = minAge;
      this.maxAge = maxAge;
      this.gender = gender;
      this.creationDate = creationDate;
    }
  };
}